controllers/installation_controller.go

```diff
@@ -32,9 +35,9 @@ const (
 type InstallationReconciler struct {
 	client.Client
 	Log              logr.Logger
-	PorterGRPCClient PorterClient
```

Comment:
Client is now not scoped to the reconciler but has a method that will create one during the invocation of the parent method.  This method makes this a little more dynamic with how we wish to test this now that we moved the instantiation to the method `CheckOrCreateInstallationOutputs`

This interface is still available, but it doesn't hang off the reconciler and is injected with patching the function we now use that is scoped to the controller.